### PR TITLE
Add capabilities

### PR DESCRIPTION
**Pull request by johndharrison**
Add options to pass additional capabilities to the desired capabilities.

### CODE REVIEW

**Comment by gaborcsardi** [comment]
Sorry, I'll look at this tomorrow. I moved it to the chromedriver branch, I can merge it into that quicker than waiting for that branch to go into master....

File: tests/testthat/test-session.R [hunk 1]
```diff
@@ -208,3 +208,24 @@ test_that("logs", {
   expect_match(log$message[1], "Hello again")
   expect_equal(names(log), c("timestamp", "level", "message"))
 })
+
+
+test_that("can pass additional parameters", {
+  user_agent_string <- "my_user_agent"
+  ap <- list(phantomjs.page.settings.userAgent = unbox("my_user_agent"))
+  s <- session$new(port = phantom$port
+                   , additional_parameters = ap
```
Comment (gaborcsardi) [inline tests/testthat/test-session.R]: Can you please reformat this like this:
```r
s <- session$new(port = phantom$port,  additional_parameters = ap)
```
or if it longer than 75 characters, like this:
```r
s <- session$new(
  port = phantom$port,
  additional_parameters = ap
)
```

and also the ones below? I know that leading comma is common is JS, but not so much in R, and I don't do it in this package. Thanks!

**Comment by johndharrison** [comment]
I added an unbox_list function which would unbox atomic elements of a list that are not marked `AsIs` or already unboxed. devtools::document() has been ran and 75 character width no leading comma formatting implemented.

File: R/session.R [hunk 2]
```diff
@@ -301,24 +304,27 @@ session <- R6Class(
 
 #' @importFrom jsonlite unbox
 
-session_initialize <- function(self, private, host, port) {
+session_initialize <- function(self, private, host, port,
+                               additional_parameters = NULL) {
```
Comment (gaborcsardi) [inline R/session.R]: No default value here please, it is not used, anyway.

File: tests/testthat/test-session.R [hunk 3]
```diff
@@ -208,3 +208,23 @@ test_that("logs", {
...
 })
+
+
+test_that("can pass additional parameters", {
+  user_agent_string <- "my_user_agent"
+  ap <- list(phantomjs.page.settings.userAgent = unbox("my_user_agent"))
+  s <- session$new(port = phantom$port, additional_parameters = ap)
+  on.exit(s$delete(), add = TRUE)
+  uastr <- "phantomjs.page.settings.userAgent"
+  expect_identical(s$.__enclos_env__$private$parameters[[uastr]],
```
Comment (gaborcsardi) [inline tests/testthat/test-session.R]: I think we don't need the hack here, the parameters should be part of `get_status()`.

File: tests/testthat/test-utils.R [hunk 4]
```diff
@@ -0,0 +1,14 @@
+
+context("utils")
+
+test_that("can unbox a list", {
+  test_list <- list(a = list(c = "un_box2", 
+                        d = list(e = c("boxed2", "boxed3")
+                                 ,f = I("boxed1"), g = "un_box3")
```
Comment (gaborcsardi) [inline tests/testthat/test-utils.R]: Another leading comma.

File: tests/testthat/test-session.R [hunk 5]
```diff
@@ -208,3 +208,23 @@ test_that("logs", {
...
+  
+})
+
+test_that("passing non list additonal paramters throws exception", {
+  expect_error(
+    s <- session$new(port = phantom$port,
+                     additional_parameters = "some string")
+  )
+  
+})
```
Comment (gaborcsardi) [inline tests/testthat/test-session.R]: Trailing ENTER missing.

File: tests/testthat/test-utils.R [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  )
+  , b = "un_box1", h = unbox("un_box4"))
+  jl_output <- toJSON(test_list, auto_unbox = TRUE)
+  ul_output <- toJSON(unbox_list(test_list))
+  expect_identical(ul_output, jl_output)
+})
+
```
Comment (gaborcsardi) [inline tests/testthat/test-utils.R]: Trailing ENTER missing.

**Pull request merged**